src/adaptors/hyperdrive/index.js

```diff
@@ -0,0 +1,324 @@
+    });
+
+    // Get token addresses and fetch prices
+    let priceWithBase = false;
+    let tokenAddress;
+    await Promise.all(poolConfig.map(async config => {
+      tokenAddress = config.vaultSharesToken === "0x0000000000000000000000000000000000000000" 
+        ? config.baseToken 
+        : config.vaultSharesToken;
+      let priceKey = `${chain}:${tokenAddress}`;
+      config.token_contract_address = tokenAddress;
+      let priceResponse = await axios.get(`https://coins.llama.fi/prices/current/${priceKey}`);
+      let price = priceResponse.data.coins[priceKey];
+      if (price === undefined && config.baseToken !== '0x0000000000000000000000000000000000000000') {
+        tokenAddress = config.baseToken;
+        priceKey = `${chain}:${tokenAddress}`;
+        priceResponse = await axios.get(`https://coins.llama.fi/prices/current/${priceKey}`);
+        price = priceResponse.data.coins[priceKey];
+        config.token_contract_address = config.baseToken;
+        priceWithBase = true;
+      }
+      // store price in config
+      config.token = price;
+      config.token.priceWithBase = priceWithBase;
+      config.token.address = tokenAddress;
+    }));
+
+    const poolInfos = (
+      await sdk.api.abi.multiCall({
+        abi: GET_POOL_INFO_ABI,
+        calls: instances.map(i => ({ target: i })),
+        chain
+      })
+    ).output.map(o => o.output);
+
+    const pools = poolNames.map((name, i) => ({ name, config: poolConfig[i], info: poolInfos[i], address: instances[i] }))
+
+    const poolsData = await Promise.allSettled(
+      pools.map(async (pool) => {
+        try {
+          const effective_share_reserves = pool.info.shareReserves - pool.info.shareAdjustment;
+          const ratio = (pool.config.initialVaultSharePrice / 1e18 * effective_share_reserves) / pool.info.bondReserves;
+          const spot_price = Math.pow(ratio, pool.config.timeStretch / 1e18);
+          const time_stretch = pool.config.positionDuration / (365 * 24 * 60 * 60);
+          const apr = (1 - spot_price) / (spot_price * time_stretch);
+
+          // time_stretch is in fractions of a year so we can use it to convert from apr to apy
+          // compounding happens every time_stretch years, so we use discrete compounding formula
+          const apy = Math.pow(1 + apr * time_stretch, 1/time_stretch) - 1;
+          const vaultSharesBalance = await queryPoolHoldings(pool, pool.config, pool.name);
+
+          // in Hyperdrive, totalSupply and tvlUsd are the same because there is no borrowing
+          let totalSupplyUsd = ((Number(vaultSharesBalance) || 0) / 10 ** pool.config.token.decimals) * pool.config.token.price;
+          // apply vaultSharePrice from config if priceWithBase is true
+          if (pool.config.token.priceWithBase) {
+            totalSupplyUsd = totalSupplyUsd * pool.info.vaultSharePrice / 1e18;
+          }
+          let tvlUsd = Number(totalSupplyUsd) || 0;
+          let totalBorrowUsd = 0;
```

Comment:
nit: turn into function

Reply:
done in 4266b9441a2b7a579c0aa09b8c001b414db78b4f and 8aed916f3d83778713c83271ec96cef29bb3d578 (remove totalsupply and totalborrow, i checked non-lending protocols don't report these)